proposals/module-types/Explainer.md

```diff
@@ -0,0 +1,725 @@
+This explainer introduces the Module Imports proposal, which allows WebAssembly
+modules to define, import and export modules and instances.
+
+1. [Problem](#problem)
+2. [Use Cases](#use-cases)
+3. [Additional Requirements](#additional-requirements)
+4. [Proposed Additions](#proposed-additions)
+   1. [Single-level Imports](#single-level-imports)
+   2. [Module and Instance Types](#module-and-instance-types)
+   3. [Instance Imports](#instance-imports)
+   4. [Module Imports and Nested Instances](#module-imports-and-nested-instances)
+   5. [First-class Definition References](#first-class-definition-references)
+   6. [Nested Modules](#nested-modules)
+   7. [Module and Instance Exports](#module-and-instance-exports)
+   8. [Summary](#summary)
+5. [Use Cases Revisited](#use-cases-revisited)
+6. [Additional Requirements Revisited](#additional-requirements-revisited)
+7. [FAQ](#faq)
+
+
+## Problem
+
+Currently, WebAssembly modules have no way to define how they are to be
+instantiated and linked together without relying on host-specific conventions.
+Consequently, the only portable way to link modules today is to [statically link]
+them, in a language/toolchain-specific manner which prevents modular code
+reuse, leads to code duplication in production and keeps languages separate.
+We would like to enable a portable, host- and language-independent ecosystem of
+composable WebAssembly modules.
+
+
+## Use Cases
+
+To motivate the proposed solution, we consider 2 use cases whose requirements
+aren't satisfied by simpler solutions. After the proposal is introduced,
+the use cases are [revisited](#use-cases-revisited) with worked-out examples
+using the proposal.
+
+
+### Link-time Virtualization
+
+When using a first-class instantiation API like the JS API's [`instantiate`],
+the imports of the module-to-be-instantiated appear as explicit arguments
+supplied by the caller. This has several useful properties that should be
```

Comment:
You're right that that is the central distinction.  To try to give a better picture of this up front, I rewrote the intro paragraph in the ## Proposed Additions section.  Maybe it would've helped clarify things for you?  Happy to have feedback and iterate on it with you.